hera_mc/cm_utils.py

```diff
@@ -157,6 +159,36 @@ def _get_astropytime(_date, _time=0):
     return return_date
 
 
+def put_keys_in_numerical_order(keys):
+    """
+    Takes a list of hookup keys in the format of prefix+number:revision and puts them in number order.
+    Returns the ordered list of keys
+    """
+    keylib = {}
+    for k in keys:
+        colon = k.find(':')
+        for i in range(len(k)):
+            try:
+                n = int(k[i:colon])
+                break
+            except ValueError:
+                continue
+        if n in keylib.keys():
+            dup_key = keylib[n][0] + str(n) + keylib[n][1]
+            print('Duplicated number found while trying to put keys in numerical order ({}, {})'.format(k, dup_key))
+            print('Returning original order')
+            return keys
+        keylib[n] = [k[:i], k[colon:]]
+    if not len(keylib.keys()):
+        return keys
+    sortkn = sorted(keylib.keys())
+    keyordered = []
+    for n in sortkn:
```

Comment:
You could avoid the temporary variable here and just write `for n in sorted(keylib.keys()):`.

Even more compact would be replacing this whole final stanza with `return [keylib[n][0] + str(n) + keylib[n][1] for n in sorted(keylib.keys())` but it would be reasonable to find that a bit too terse.